### PR TITLE
core/clist: add clist_sort()

### PR DESCRIPTION
**Pull request by kaspar030**
This PR adds a function that sorts clists, using a stable merge-sort algorithm that runs in O(N log N) best-case and worst-case.

The actual sorting code is BSD-licensed (taken from https://www.chiark.greenend.org.uk/~sgtatham/algorithms/listsort.html).

### CODE REVIEW

File: core/clist.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/**
+ * @ingroup     core_util
+ * @{
+ *
+ * @file
+ * @brief       clist helper implentations
```
Comment (smlng) [inline core/clist.c]: /implentations/implementations/

File: core/clist.c [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        nmerges = 0;  /* count number of merges we do in this pass */
+
+        while (p) {
+            nmerges++;  /* there exists a merge to be done */
+            /* step `insize' places along from p */
+            q = p;
+            psize = 0;
+            for (i = 0; i < insize; i++) {
+                psize++;
+                q = (q->next == oldhead ? NULL : q->next);
```
Comment (smlng) [inline core/clist.c]: I would place parentheses around the condition, i.e., `(q->next == oldhead)`

File: core/clist.c (hunk 1 above)
Comment (kaspar030) [reply core/clist.c]: fixed

File: core/clist.c (hunk 2 above)
Comment (kaspar030) [reply core/clist.c]: Yup.

**Comment by smlng** [comment]
Please fix issues reported by Murdock, and also squash - pre-ACK.

**Comment by gebart** [comment]
Sorry, I don't have time to review this at the moment.

**Comment by kaspar030** [comment]
> Please fix issues reported by Murdock

done

**Comment by OlegHahm** [comment]
> kaspar030 added Ready for CI build Ready for CI build Ready for CI build and removed Ready for CI build Ready for CI build Ready for CI build labels 5 hours ago

lol

What is the motivation for this PR?

**Comment by kaspar030** [comment]
> What is the motivation for this PR?

Needed this for a pet project...

Also I'm trying to fill the feature gap between clist & utlist. (#6209)

**Comment by kaspar030** [comment]
> I think all comments were addressed. @smlng you good?

*ping* might have slipped through the release already...

**Comment by kaspar030** [comment]
@smlng *ping* you've basically already ACK'ed this. :)

**Comment by smlng** [comment]
ah sorry, for the delay ...

**Comment by kaspar030** [comment]
&go.

Thanks @smlng!

**Pull request merged**